core/eth/state.go

```diff
@@ -59,7 +85,131 @@ func (cs *ChainState) GetOperatorSocket(ctx context.Context, blockNumber uint, o
 	return socket, nil
 }
 
-func getOperatorState(operatorsByQuorum core.OperatorStakes, blockNumber uint32) (*core.OperatorState, error) {
+// buildSocketMap returns a map from operatorID to socket address for the operators in the operatorsByQuorum
+func (cs *ChainState) buildSocketMap(ctx context.Context, operatorIds []core.OperatorID) error {
+	socketMap := make(map[core.OperatorID]*string)
+	for _, operatorID := range operatorIds {
+		// if the socket is already in the map, skip
+		if _, ok := socketMap[operatorID]; ok {
+			continue
+		}
+		socket, err := cs.Tx.GetOperatorSocket(ctx, operatorID)
+		if err != nil {
+			return err
+		}
+		socketMap[operatorID] = &socket
+	}
+
+	cs.socketMu.Lock()
+	for operatorID, socket := range socketMap {
+		cs.SocketMap[operatorID] = socket
+	}
+	cs.socketMu.Unlock()
+
+	return nil
+}
+
+// indexSocketMap indexes a block range of socket update events
+func (cs *ChainState) indexSocketMap(ctx context.Context) error {
+	currentBlockNumber, err := cs.GetCurrentBlockNumber()
```

Comment:
Also something to consider, but doesn't need to be changed in this PR if you prefer not to:

`GetCurrentBlockNumber` is getting the block number by calling `cs.Client.HeaderByNumber`, and then parsing the block number out of the header. I think it could simply call `cs.Client.BlockNumber` instead